### PR TITLE
To determine the name of the processor

### PR DESCRIPTION
**Pull request by GitScrum**
It is necessary to determine the exact name of the processor.
Now it's not obvious - https://github.com/post-org/post-load-plugins/blob/master/src/post-processor.js

### CODE REVIEW

**Comment by GitScrum** [comment]
@posthtml/collaborators ping.

File: lib/index.js [hunk 1]
```diff
@@ -89,6 +103,7 @@ PostHTML.prototype.process = function (tree, options) {
   tree = options.skipParse ? tree : parser(tree)
 
   tree.options = options
+  tree.processor = this
```
Comment (michael-ciniawsky) [inline lib/index.js]: Would the name && version really be needed on the tree? 

```js
- tree.processor = this

+ tree.name = this.name
+ tree.version = this.version
```

File: lib/index.js [hunk 2]
```diff
@@ -19,6 +19,20 @@ var render = require('posthtml-render')
...
+  /**
+   * @example
+   * if ( ctx.processor.version.split('.')[1] !== '9' ) {
+   *   throw new Error('This plugin works only with PostHTML 9');
+   * }
+   */
+  this.version = '0.9.1'
+
+  /**
+   * @example
```
Comment (michael-ciniawsky) [inline lib/index.js]: Same here :)

File: lib/index.js [hunk 3]
```diff
@@ -19,6 +19,20 @@ var render = require('posthtml-render')
  */
 function PostHTML (plugins) {
   this.plugins = plugins || []
+
+  /**
+   * @example
```
Comment (michael-ciniawsky) [inline lib/index.js]: `post-load-*` or any other plugin are unrelated to posthtml core docs and should be documented/mentioned elsewhere imho

File: lib/index.js [hunk 4]
```diff
@@ -19,6 +19,20 @@ var render = require('posthtml-render')
  */
 function PostHTML (plugins) {
   this.plugins = plugins || []
```
Comment (michael-ciniawsky) [inline lib/index.js]: Maybe better to put name && version before plugins

```js
this.name = 'posthtml'
this.version = '0.9.1'

this.plugins = plugins || []
```

File: lib/index.js (hunk 3 above)
Comment (GitScrum) [inline lib/index.js]: There is no need to refer to the name of the process if there is a plugin `posthtml`. 
This is one example of how you can use

File: lib/index.js (hunk 1 above)
Comment (GitScrum) [inline lib/index.js]: > Would the name && version really be needed on the tree?

Your options to reach process ?

```js
+ tree.processor = this
```

This should be in the constructor/processor `ctx.processor.name`
```js
- tree.name = this.name
- tree.version = this.version
```

Comment (michael-ciniawsky) [inline lib/index.js]:
```js
tree.processor = { name: this.name, version: this.version }
tree.plugins = this.plugins
```

But plugins must be on the processor, for your usecase ?

File: lib/index.js (hunk 3 above)
Comment (michael-ciniawsky) [inline lib/index.js]: Yep sure, but external modules/ plugins in general should not be part of the core docs when possible :)

```js
/**
 * PostHTML Instance
 * 
 * @name posthtml
 * @version 0.9.1
 * 
 * @prop plugins
 * @prop options
 */
this.name = 'posthtml'
this.version = '0.9.1'

this.plugins = plugins || []
this.options = {}
```

File: lib/index.js (hunk 1 above)
Comment (GitScrum) [inline lib/index.js]: Why is this determined in the process ? If the property list will be expanded ? if I want to instantiate a new object ? This should be in the constructor gives you more flexibility.

```js
tree.processor.constructor.name ???
```

File: lib/index.js (hunk 3 above)
Comment (GitScrum) [inline lib/index.js]:
```js
this.options = {}
```
it is not necessary as [determined in the process](https://github.com/GitScrum/posthtml/blob/master/lib/index.js#L74-L91)

File: lib/index.js [hunk 5]
```diff
@@ -18,6 +18,17 @@ var render = require('posthtml-render')
...
+  /**
+ * PostHTML Instance
+ *
+ * @name posthtml
+ * @version 0.9.1
+ *
+ * @prop plugins
+ * @prop options
+ */
+  this.version = '0.9.1'
```
Comment (voischev) [inline lib/index.js]: mb `require('../package.json').{name|version}` ???

File: lib/index.js [hunk 6]
```diff
@@ -18,6 +18,17 @@ var render = require('posthtml-render')
  * @param {Array} plugins - An array of PostHTML plugins
  */
 function PostHTML (plugins) {
+  /**
+ * PostHTML Instance
+ *
+ * @name posthtml
+ * @version 0.9.1
```
Comment (voischev) [inline lib/index.js]: mb delete this line or autoupdating script?

File: lib/index.js (hunk 5 above)
Comment (voischev) [inline lib/index.js]: hmm... require cache?

Comment (mrmlnc) [inline lib/index.js]: Modules are cached after the first time they are loaded. But before return an object - Node.js will calculate the full path (module id) to `require()`. For more details, we can trace the call chain:

  1. https://github.com/nodejs/node/blob/master/lib/module.js#L498
  2. https://github.com/nodejs/node/blob/master/lib/module.js#L418
  3. https://github.com/nodejs/node/blob/master/lib/module.js#L461
  4. https://github.com/nodejs/node/blob/master/lib/module.js#L468
  5. etc.

P.S. It's a small delay.

Comment (voischev) [inline lib/index.js]: I only care about the automation increment version of this place...

Comment (GitScrum) [inline lib/index.js]: @voischev the first interpolation. i have create new issue when will configured build process with babili, babel and there will be to write a plugin to replace the version.

Comment (voischev) [inline lib/index.js]: Oh. let no Babel please?! simple `require` a priority

File: lib/index.js [hunk 7]
```diff
@@ -8,7 +9,7 @@ var render = require('posthtml-render')
  *         Anton Winogradov (@awinogradov),
  *         Alexej Yaroshevich (@zxqfox),
  *         Vasiliy (@Yeti-or)
- * @version 0.9.0
+ * @version 0.9.1
```
Comment (voischev) [inline lib/index.js]: del this line to

File: lib/index.js (hunk 5 above)
Comment (GitScrum) [inline lib/index.js]: > Babel mb letter ok?
Ok :)

**Comment by GitScrum** [comment]
@michael-ciniawsky approve ?

**Comment by michael-ciniawsky** [comment]
@GitScrum  yep, sry I forgot :)

**Comment by GitScrum** [comment]
@voischev merge ?

**Comment by GitScrum** [comment]
@voischev npm publish ?

**Comment by voischev** [comment]
@GitScrum waiting for #200?

**Pull request merged**